draft-enghardt-panrg-path-properties.md

```diff
@@ -46,8 +48,21 @@ This information is expressed as properties of paths between two endpoints.
 # Introduction
 
 Because the current Internet provides an IP-based best-effort bit pipe, endpoints have little information about paths to other endpoints.
-A Path Aware Network exposes information about one or multiple paths through the network to endpoints,
-so that endpoints can use this information.
+A Path Aware Network exposes information about one or multiple paths through the network to endpoints or the network infrastructure.
+
+It is impossible to provide an exhaustive list of path properties, as with every new technology and protocol, novel properties might become relevant.
+In order to specify broadly relevant path properties, scenarios in which knowledge about (additional) path information gives advantages are described here.
+
+Traffic policies leveraging path awareness can be defined.
+Such policies can allow or disallow sending traffic over specific networks or nodes, select an appropriate protocol depending on the capabilities of the on-path devices (e.g., using QUIC instead of TCP if no device blocks UDP traffic), or adjust protocol parameters to the used path.
```

Comment:
I'm not quite sure I understand this use case yet, and how it relates to the knowledge about certain path properties.
Maybe it would help to give examples of who might define such traffic policies (network operators? end users?) and what path properties they might use. I think it's better to use active voice than passive voice anyway ("can be defined" --> "network operators or end users can define ...").

In the example given here, I just realized that we have this as a property (whether a certain transport protocol works or not), but it's not like the devices on the path tell us whether they block QUIC -- an endpoint would just have to try a QUIC handshake and fall back to TCP if it doesn't work, right? Then the endpoint could cache the "Transport protocols available" property with its knowledge of whether the endpoint has established a QUIC connection successfully in the past (certain time period), and then the endpoint could be "more confident" in just using QUIC, i.e., it could put less resources into trying out different protocols. But to me, this is not really a policy or a matter of allowing or not allowing traffic, but more, like, an opportunity for performance optimization.

Maybe we can find a more obvious example of an actual policy here.
I'm thinking along the lines of:  An end user might define a policy that they prefer their traffic to pass through a specific network, e.g., their home network rather than a public cellular network. To inform such a policy, it might be useful for the end user to know about properties of the paths such as Monetary Cost or the presence of a specific network on the path (i.e., a fixed-line network where they have a flat-rate contract with the provider, as opposed to a cellular network where the provider charges based on data traffic volume).

Maybe we can also find a policy use case with firewalls or proxies, but I'm not sure if those are easier to understand and agree with for our audience.

Reply:
I'll change to active voice.

Maybe blocking UDP traffic is not a good example. Should we use the following examples (one for each of the use cases):

- An endpoint might require either an on-path or locally installed firewall to send or receive traffic.

- Endpoints communicating with each other within a trusted network might forgo encryption, such as TLS, as long as all devices and links are within the network.

- A video streaming application chooses an (initial) video quality based on the achievable data rate, or the monetary cost of the link, whether path uses a volume-based cost model or a flat-rate model.

Although I'm not too sure about the firewall & encryption example as some people might be hesitant to use path properties for security purposes (is the entity providing path properties trusted? Are path properties authenticated? Are the capabilities of each device authenticated?)...

Reply:
All of these examples are about policy, i.e., the first use case, right? 
Yes, I'm not sure about using security examples either, if we do not have anything on trustworthiness path properties yet (which is out of scope for our draft, but still, it would bother me as well). 
However, maybe we could use the first example, e.g., say that in an enterprise network there is a policy that all traffic has to go through a firewall, and then the endpoint wants to be aware of that... about the locally-installed firewall, I'm not sure what it has to do with path properties, as it's all local.
The third example is a good convincing one to me.

I just realized that these examples are also about path selection, just like the name of our third use case, so maybe we have to name the third use case not "path selection" but "traffic optimization" or something similar, to highlight that the third use case is about resource usage and/or performance, not policy.

Reply:
Sorry for the confusion. What I meant to say is they are for different aspect of the first use case "traffic policies".

I think "traffic policies" is about path transparency (knowledge about existing paths) rather than path choice (which is handled in the third use case), since we always work with (possibly one) existing path. I would rather keep the title of the third use case as "path selection", as our first use case also has aspects of "traffic optimization".

An example of a traffic policy is a video streaming application choosing an (initial) video quality based on the achievable data rate, or the monetary cost of the link using a volume-based or flat-rate cost model.
Another example is an enterprise network where all traffic has to go through a firewall, in which case the endpoint needs to be aware of on-path firewalls.

Reply:
Yep, I think we can use these examples like that.
And yes, I agree to leave path selection separate from "traffic policies".